### PR TITLE
Celsius temperature for freezing point

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary

SUMMARY: Infrastructure "Use celsius instead of fahrenheit for freezing point in json"

#### Purpose of change

Most of the json thing use metric. 
Temperature is not in metric. It is in fahrenheit.
It would be better for it to use celsius.

#### Describe the solution

* "freezing_point" attribute for items is now in celsius.
* "freeze_point" attribute for materials was renamed to "freezing_point" to be same as for items. It is also in celsius now. It is not actually used anywhere but at least it is consistent.

Both of them are also now floats. So if you want to set the freezing temperature precisely you can use decimals.

#### Describe alternatives you've considered

Let it be in fahrenheit.
Use Kelvins instead of celsius.

Freezing point is always used as kelvins in code. The celsius could be converted to kelvin when the json is loaded.

#### Testing

Tests pass. 
Freezing points seem to be what they should be.
Using decimal freezing point works.

#### Additional context

The item freezing points could be audited a bit. Alcohols have lots of variation in their freezing points for no apparent reason.

The way material freezing point works could be reworked. Currently it is can be used only for corpses (but it is not used even there). It would probably be good for items to fall back to material freezing point if the item has no defined freezing point.

### CODE REVIEW

**Comment by actual-nh** [comment]
Are temperatures used elsewhere in the code (e.g., weather)? Which is used there?

**Comment by anothersimulacrum** [comment]
Maybe we should have units (units.h) for temperature?

**Comment by Hirmuolio** [comment]
Item temperature is in Kelvins.

Rotting is in fahrenheit.
Weather is in fahrenheit AFAIK.
I have not looked at player body temperature things.

**Comment by actual-nh** [comment]
> Item temperature is in Kelvins.

This would appear to be an argument for using Kelvins for freezing/melting/boiling points (unless it gets changed).

File: src/item.cpp [hunk 1]
```diff
@@ -1777,7 +1777,7 @@ void item::debug_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
                                           get_latent_heat() ) );
                 info.push_back( iteminfo( "BASE", _( "Freeze point: " ), "",
                                           iteminfo::lower_is_better,
-                                          get_freeze_point() ) );
+                                          get_freeze_point() * 1000 ) );
```
Comment (BrettDong) [inline src/item.cpp:1780]: Why multiply 1000 here?

Comment (Hirmuolio) [reply src/item.cpp:1780]: The debug info only shows integers. So multiplying by 1000 makes the decimal part visible.
The specific heats before it do the same.
It is a bit dirty thing to do. But I think it is good enough for debug info. If someone knows how to show decimals there that can be put there instead.

Comment (BrettDong) [reply src/item.cpp:1780]: 螢幕截圖 2021-01-26 下午3 34 48

```diff
diff --git a/src/item.cpp b/src/item.cpp
index 2877abbce9..3c49541603 100644
--- a/src/item.cpp
+++ b/src/item.cpp
@@ -1776,8 +1776,8 @@ void item::debug_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
                                           iteminfo::lower_is_better,
                                           get_latent_heat() ) );
                 info.push_back( iteminfo( "BASE", _( "Freeze point: " ), "",
-                                          iteminfo::lower_is_better,
-                                          get_freeze_point() * 1000 ) );
+                                          iteminfo::lower_is_better | iteminfo::is_decimal,
+                                          get_freeze_point() ) );
```

**Pull request merged**